library/src/main/scala/scala/scalajs/runtime/RuntimeString.scala

```diff
@@ -166,6 +166,32 @@ private[runtime] object RuntimeString {
     Pattern.matches(regex, thiz)
   }
 
+  // Both regionMatches ported from https://github.com/gwtproject/gwt/blob/master/user/super/com/google/gwt/emul/java/lang/String.java
+  def regionMatches(thiz: String, ignoreCase: Boolean,
+                    toffset: Int, other: String,
+                    ooffset: Int, len: Int): Boolean = {
+    if (other == null) {
+      throw new NullPointerException()
+    }
+    else if (toffset < 0 || ooffset < 0 || len <= 0) {
```

Comment:
If `len` is negative, you must return true in some cases. Have a look at the [JavaDoc][1].

[1]: http://docs.oracle.com/javase/7/docs/api/java/lang/String.html#regionMatches(boolean,%20int,%20java.lang.String,%20int,%20int)